backend/core/tests/test_quivr_rag.py

```diff
@@ -2,37 +2,41 @@
 
 import pytest
 from quivr_core.chat import ChatHistory
-from quivr_core.config import LLMEndpointConfig, RAGConfig
+from quivr_core.config import LLMEndpointConfig, RetrievalConfig
 from quivr_core.llm import LLMEndpoint
 from quivr_core.models import ParsedRAGChunkResponse, RAGResponseMetadata
-from quivr_core.quivr_rag import QuivrQARAG
+from quivr_core.quivr_rag_langgraph import QuivrQARAGLangGraph
 
 
 @pytest.fixture(scope="function")
 def mock_chain_qa_stream(monkeypatch, chunks_stream_answer):
     class MockQAChain:
-        async def astream(self, *args, **kwargs):
+        async def astream_events(self, *args, **kwargs):
             for c in chunks_stream_answer:
-                yield c
+                yield {
+                    "event": "on_chat_model_stream",
+                    "metadata": {"langgraph_node": "generate"},
+                    "data": {"chunk": c},
+                }
 
     def mock_qa_chain(*args, **kwargs):
         return MockQAChain()
 
-    monkeypatch.setattr(QuivrQARAG, "build_chain", mock_qa_chain)
+    monkeypatch.setattr(QuivrQARAGLangGraph, "build_chain", mock_qa_chain)
 
 
 @pytest.mark.base
 @pytest.mark.asyncio
-async def test_quivrqarag(
+async def test_quivrqaraglanggraph(
     mem_vector_store, full_response, mock_chain_qa_stream, openai_api_key
 ):
     # Making sure the model
     llm_config = LLMEndpointConfig(model="gpt-4o")
     llm = LLMEndpoint.from_config(llm_config)
-    rag_config = RAGConfig(llm_config=llm_config)
+    retrieval_config = RetrievalConfig(llm_config=llm_config)
     chat_history = ChatHistory(uuid4(), uuid4())
-    rag_pipeline = QuivrQARAG(
-        rag_config=rag_config, llm=llm, vector_store=mem_vector_store
+    rag_pipeline = QuivrQARAGLangGraph(
```

Comment:
I agree, I think that this will evolve once we clarity / implement a ConfigService and clarify the whole architecture... I'd keep it like this for now